### PR TITLE
Bump up the version for ROCm-5.7.0 and ROCm-5.7.1 releases.

### PR DESCRIPTION
**Pull request by srekolam**
initial commit changes to the rocm recipes to support the ROCm-5.7.0 and ROCm-5.7.1 releases

### CODE REVIEW

File: var/spack/repos/builtin/packages/hiprand/package.py
Comment (cgmb) [inline var/spack/repos/builtin/packages/hiprand/package.py:25]: This is not the right commit. I've pushed a tag for ROCm 5.7.1.

**Comment by rbberger** [comment]
@srekolam you probably also want to add `gfx941` and `gfx942` in https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/rocm.py#L110 along with the appropriate lines for `llvm-amdgpu` here https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/rocm.py#L170

**Comment by srekolam** [comment]
@spackbot rerun pipeline

**Comment by srekolam** [comment]
@spackbot rerun pipeline

**Comment by srekolam** [comment]
@spackbot rerun pipeline

**Comment by bvanessen** [comment]
@srekolam Can we address issues with the use of HIP_PATH in this PR?  Also what is holding this PR up from merging?

**Comment by srekolam** [comment]
> @srekolam Can we address issues with the use of HIP_PATH in this PR? Also what is holding this PR up from merging?

@bvanessen , we are waiting for any review comments to be addressed for this PR. For the HIP_PATH issues that you have mentioned can you kindly let us know if there are any spack issues that are newly open or can create one ? we will fix them in the next PR quickly.

**Comment by bvanessen** [comment]
> > @srekolam Can we address issues with the use of HIP_PATH in this PR? Also what is holding this PR up from merging?
> 
> @bvanessen , we are waiting for any review comments to be addressed for this PR. For the HIP_PATH issues that you have mentioned can you kindly let us know if there are any spack issues that are newly open or can create one ? we will fix them in the next PR quickly.

@srekolam There is a bug that we are tracking with AMD that states that this variable should not be used in newer versions, but more importantly in 5.7.x it is pointing to the wrong place.

**Review by afzpatel — APPROVED** [review]

**Review by renjithravindrankannath — APPROVED** [review]

**Comment by bvanessen** [comment]
> > > @srekolam Can we address issues with the use of HIP_PATH in this PR? Also what is holding this PR up from merging?
> > 
> > 
> > @bvanessen , we are waiting for any review comments to be addressed for this PR. For the HIP_PATH issues that you have mentioned can you kindly let us know if there are any spack issues that are newly open or can create one ? we will fix them in the next PR quickly.
> 
> @srekolam There is a bug that we are tracking with AMD that states that this variable should not be used in newer versions, but more importantly in 5.7.x it is pointing to the wrong place.

@srekolam I think that if the `HIP_PATH` is set to `ROCM_PATH` for 5.7.x that will address the bug.

**Comment by srekolam** [comment]
> > > > @srekolam Can we address issues with the use of HIP_PATH in this PR? Also what is holding this PR up from merging?
> > > 
> > > 
> > > @bvanessen , we are waiting for any review comments to be addressed for this PR. For the HIP_PATH issues that you have mentioned can you kindly let us know if there are any spack issues that are newly open or can create one ? we will fix them in the next PR quickly.
> > 
> > 
> > @srekolam There is a bug that we are tracking with AMD that states that this variable should not be used in newer versions, but more importantly in 5.7.x it is pointing to the wrong place.
> 
> @srekolam I think that if the `HIP_PATH` is set to `ROCM_PATH` for 5.7.x that will address the bug.

@bvanessen , i just aded https://github.com/spack/spack/pull/40724/commits/b816c25a0474dfdc031ba470e5ffd91c386a4293 
can you please let me know if this will address the bug ?

**Comment by bvanessen** [comment]
> > > > > @srekolam Can we address issues with the use of HIP_PATH in this PR? Also what is holding this PR up from merging?
> > > > 
> > > > 
> > > > @bvanessen , we are waiting for any review comments to be addressed for this PR. For the HIP_PATH issues that you have mentioned can you kindly let us know if there are any spack issues that are newly open or can create one ? we will fix them in the next PR quickly.
> > > 
> > > 
> > > @srekolam There is a bug that we are tracking with AMD that states that this variable should not be used in newer versions, but more importantly in 5.7.x it is pointing to the wrong place.
> > 
> > 
> > @srekolam I think that if the `HIP_PATH` is set to `ROCM_PATH` for 5.7.x that will address the bug.
> 
> @bvanessen , i just aded [b816c25](https://github.com/spack/spack/commit/b816c25a0474dfdc031ba470e5ffd91c386a4293) can you please let me know if this will address the bug ?

@srekolam That should do the trick.

**Comment by rbberger** [comment]
Would be great if we had this in Spack `develop` for the COE hackathon next week.

**Comment by srekolam** [comment]
@tldahlgren can you please review and let me know if this can be merged?

**Review by afzpatel — APPROVED** [review]

**Review by cgmb — APPROVED** [review]
The canonical URLs for these libraries have all changed. They've been moved to the https://github.com/ROCm organization. However, the old URLs should still work.

File: lib/spack/spack/build_systems/rocm.py
Comment (cgmb) [inline lib/spack/spack/build_systems/rocm.py]: The gfx940 target was added to llvm-amdgpu in ROCm 5.2. I don't think that needs to be changed. We may also want to add the xnack specializations for those targets. The only reason the AMD binary packages didn't use `xnack-` and `xnack+` for gfx94x was because the binary bloat would be too great. Spack users don't have to build for all architectures, so that specialization _could_ be useful to them.

**Comment by srekolam** [comment]
tried adding xnack- and xnack+ to gfx94x but the prechecks were failing.

**Review by tldahlgren — APPROVED** [review]
Confirmed the new version directive sha256s and all sha256 for `rpp`.

**Comment by tldahlgren** [comment]
@v-dobrev Do you want to comment on the `mesa` package changes here before this PR is merged?

**Comment by v-dobrev** [comment]
> @v-dobrev Do you want to comment on the `mesa` package changes here before this PR is merged?

They look good to me.

**Review by v-dobrev — APPROVED** [review]

**Pull request merged**
